src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,170 @@
+                            Map.entry("getTime(int)", rs -> rs.getTime(1)),
+                            Map.entry("getTime(int,Calendar)", rs -> rs.getTime(1, Calendar.getInstance())),
+                            Map.entry("getTimestamp(int)", rs -> rs.getTimestamp(1)),
+                            Map.entry("getTimestamp(int,Calendar)", rs -> rs.getTimestamp(1, Calendar.getInstance())),
+                            Map.entry("getBigDecimal(int)", rs -> rs.getBigDecimal(1)),
+                            Map.entry("getArray(int)", rs -> rs.getArray(1)),
+                            Map.entry("getObject(int,Class)", rs -> rs.getObject(1, String.class)),
+                            Map.entry("getMetaData()", ResultSet::getMetaData),
+                            Map.entry("findColumn(String)", rs -> rs.findColumn("name")),
+                            Map.entry("isWrapperFor(Class)", rs -> rs.isWrapperFor(MongoResultSet.class)))
+                    .entrySet()
+                    .stream()
+                    .map(entry -> Arguments.of(entry.getKey(), entry.getValue()));
+        }
+    }
+
+    @Nested
+    class ColumnIndexCheckingTests {
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too low")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexUnderflow")
+        void testColumnIndexUnderflow(String label, ResultSetMethodInvocation methodInvocation) {
+            // when && then
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too high")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexOverflow")
+        void testColumnIndexOverflow(String label, ResultSetMethodInvocation methodInvocation) {
+            // when && then
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsWithColumnIndexUnderflow() {
+            return doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(0);
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsWithColumnIndexOverflow() {
+            return doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(FIELDS.size() + 1);
+        }
+
+        private static Stream<Arguments> doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(int columnIndex) {
+            return Map.<String, ResultSetMethodInvocation>ofEntries(
+                            Map.entry("getString(int)", rs -> rs.getString(columnIndex)),
```

Comment:
Let's eliminate this repetition? It seems that these all must be affected by close, and also behave in a certain way for index bounds checking.

Reply:
Closing also needs a parameter, which is just 1. Could you be more specific about why it would be clumsy or hard to read?

Reply:
I've adopted your suggestion in the latest commit. I might well apply it to existing JDBC similar testing cases as well (not in this PR).